assessment_software/src/app/file-upload-component/file-upload.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-file-upload',
+  templateUrl: './file-upload.component.html',
+  styleUrls: ['./file-upload.component.css']
+})
+export class FileUploadComponent {
+    listOfFiles:any[]=[];
+    
+    @Input() type: string = 'file';
+    @Input() accept: string = '';
+    @Input() styles: string = '';
+    @Input() fileSizeLimit:any;
+    @Input() error:boolean=false;
+    
+    handleFileSize(event:any){
+      this.listOfFiles=event.target.files;
+      for (let file of this.listOfFiles){
+        if (file.size/1000000>=this.fileSizeLimit)
+        this.error= true;
```

Comment:
For true case:

after setting this.error = true; 
setting also the error message is best. this.errorMessage = "Sorry, the file you uploaded is too large. The maximum file size is this.fileSizeLimit MB"

and for the false case:
setting the errorMessage to Unknown error which is the default error message:
"Unknown Error"